From 1f6bf0cae40c1076faa39707c56d3994e73d01e2 Mon Sep 17 00:00:00 2001
From: David Drysdale <drysdale@google.com>
Date: Wed, 27 Mar 2024 11:46:55 +0000
Subject: [PATCH] Update legacy provisioning msg list

Bug: 329701910
Test: trusty_keymaster_set_attestation_ids
Ignore-AOSP-First: security impact
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:6b764647e769c8b1bf3bb131ecc6724f4fee7a82)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:0fcb837517f73318d12c5a673ac73620a41d0d17)
Merged-In: I797438c18a9e0a9d8c26e3039805407b0c217893
Change-Id: I797438c18a9e0a9d8c26e3039805407b0c217893
---
 wire/src/legacy.rs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/wire/src/legacy.rs b/wire/src/legacy.rs
index d418f82..3cd519b 100644
--- a/wire/src/legacy.rs
+++ b/wire/src/legacy.rs
@@ -601,12 +601,15 @@ declare_req_rsp_enums! { TrustyKeymasterOperation => (TrustyPerformOpReq, Trusty
     GetVersion = 7 =>                                (GetVersionRequest, GetVersionResponse),
     GetVersion2 = 28 =>                              (GetVersion2Request, GetVersion2Response),
     SetBootParams = 0x1000 =>                        (SetBootParamsRequest, SetBootParamsResponse),
+
+    // Provisioning-related requests. Changes here should be reflected in `is_trusty_provisioning_{code,req}`.
     SetAttestationKey = 0x2000 =>                    (SetAttestationKeyRequest, SetAttestationKeyResponse),
     AppendAttestationCertChain = 0x3000 =>           (AppendAttestationCertChainRequest, AppendAttestationCertChainResponse),
     ClearAttestationCertChain = 0xa000 =>            (ClearAttestationCertChainRequest, ClearAttestationCertChainResponse),
     SetWrappedAttestationKey = 0xb000 =>             (SetWrappedAttestationKeyRequest, SetWrappedAttestationKeyResponse),
     SetAttestationIds = 0xc000 =>                    (SetAttestationIdsRequest, SetAttestationIdsResponse),
     SetAttestationIdsKM3 = 0xc001 =>                 (SetAttestationIdsKM3Request, SetAttestationIdsKM3Response),
+
     ConfigureBootPatchlevel = 0xd0000 =>             (ConfigureBootPatchlevelRequest, ConfigureBootPatchlevelResponse),
 } }
 
@@ -643,6 +646,7 @@ pub fn is_trusty_provisioning_code(code: u32) -> bool {
             | Some(TrustyKeymasterOperation::ClearAttestationCertChain)
             | Some(TrustyKeymasterOperation::SetWrappedAttestationKey)
             | Some(TrustyKeymasterOperation::SetAttestationIds)
+            | Some(TrustyKeymasterOperation::SetAttestationIdsKM3)
     )
 }
 
@@ -655,6 +659,7 @@ pub fn is_trusty_provisioning_req(req: &TrustyPerformOpReq) -> bool {
             | TrustyPerformOpReq::ClearAttestationCertChain(_)
             | TrustyPerformOpReq::SetWrappedAttestationKey(_)
             | TrustyPerformOpReq::SetAttestationIds(_)
+            | TrustyPerformOpReq::SetAttestationIdsKM3(_)
     )
 }
 
-- 
2.46.1.824.gd892dcdcdd-goog

